src/data_write.py

```diff
@@ -996,238 +766,171 @@ def find_expectation_value(x, parameters, field_name):
-            :type   parameters_holder:  dict
+            :param  aveperiod_data:  Dict that holds SequenceData for each slice.
+            :type   aveperiod_data:  dict
             """
 
-            needed_fields = [
-            "borealis_git_hash",    "experiment_id",                "experiment_name",
-            "experiment_comment",   "num_slices",                   "slice_comment",
-            "station",              "num_sequences",                "rx_sample_rate",
-            "scan_start_marker",    "int_time",                     "tx_pulse_len",
-            "tau_spacing",          "main_antenna_count",           "intf_antenna_count",
-            "freq",                 "samples_data_type",            "pulses",
-            "sqn_timestamps",       "beam_nums",                    "beam_azms",
-            "data_dimensions",      "data_descriptors",             "antenna_arrays_order",
-            "data",                 "num_samps",                    "pulse_phase_offset",
-            "noise_at_freq",        "data_normalization_factor",    "blanked_samples",
-            "slice_id",             "slice_interfacing",            "scheduling_mode",
-            "gps_locked",           "gps_to_system_time_diff",      "agc_status_word",
-            "lp_status_word"
-            ]
-
             antenna_iq = data_parsing.antenna_iq_accumulator
             slice_id_list = [x for x in antenna_iq.keys() if isinstance(x, int)]
 
-            # Pop these so we don't include them in later iteration.
-            data_descriptors = antenna_iq.pop('data_descriptors', None)
-
             # Parse the antennas from message
             rx_main_antennas = {}
             rx_intf_antennas = {}
 
-            for meta in aveperiod_meta.sequences:
-                for rx_freq in meta.rx_channels:
-                    rx_main_antennas[rx_freq.slice_id] = list(rx_freq.rx_main_antennas)
-                    rx_intf_antennas[rx_freq.slice_id] = list(rx_freq.rx_intf_antennas)
+            for sqn in aveperiod_meta.sequences:
+                for rx_channel in sqn.rx_channels:
+                    rx_main_antennas[rx_channel.slice_id] = list(rx_channel.rx_main_antennas)
+                    rx_intf_antennas[rx_channel.slice_id] = list(rx_channel.rx_intf_antennas)
 
             # Build strings from antennas used in the message. This will be used to know
             # what antennas were recorded on since we sample all available USRP channels
             # and some channels may not be transmitted on, or connected.
-            main_ant_str = lambda x: f"antenna_{x}"
-            intf_ant_str = lambda x: f"antenna_{x + self.options.main_antenna_count}"
-            for slice_id in rx_main_antennas:
-                rx_main_antennas[slice_id] = [main_ant_str(x) for x in rx_main_antennas[slice_id]]
-                rx_intf_antennas[slice_id] = [intf_ant_str(x) for x in rx_intf_antennas[slice_id]]
+            def main_ant_str(x):
```

Comment:
If you already want to toss the lambdas, why not just put the fstring right in the list comprehension?